### DISCUSSION TITLE
How do I install airflow using helm on arm architecture

### DISCUSSION THREAD

**Discussion by XiaoXin101** [apache/airflow#23778] [General]
### Official Helm Chart version

1.5.0 (latest released)

### Apache Airflow version

2.3.0 (latest released)

### Kubernetes Version

1.18.2

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
I was truly hoping people are reading the messages printed when they attempt to open an issue.
PLEASE Follow the guildelines when you are opening an issue. It's written there CLEARLY that if you have questions you should open DISCUSSION rather than ISSUE, including a link that you can click to open one.

Answering your question. You do not need to do anything. just run it on ARM. Note that ARM images are released for 2.3.0 only and that currently ARM only works for Postgres (MySQL and MSSQL had no ARM-ready drivers). 

Note that our ARM support is experimental - and we have not tested it thoroughly in production - so if you have any problems and experiences and fix them (we do not promise it will work for sure and in all configurations) - it would be great you share them here with others.

**Reply by XiaoXin101** [reply]
I mean, when I install it, it seems that what I pull is still an AMD64 image

this is error log: standard_init_linux.go:211: exec user process caused "exec format error"

**Reply by potiuk** [reply]
If you are pulling 2.3.0 image it shoudl be ARM. If any earlier image - it should be AMD. You can try to pull it manually and see if it works. It should.

Please investigate further and let us know - being eperimental, things might not work but if you investigate, do some experimenting and put some more comprehensive information, including pull logs, what you have experimented with and what exactly does not work, that would be useful for us and others.

You have not explained what exactly you have done and how you tried to understand what's going on. Please do.

**Reply by potiuk** [reply]
You. have not written for example which image you had problem with. Please be more verbose in describing your problem if you really want - first of all - to help others understand what you do and how you tried to investigate problem so far, secondly to share your experiment results with others, and thirdly to give a chance anyone to help you. For now what I hear is

"things do not work"

Bear in mind that this community works on a basis of self-help - you need to show that you've done your part and rather than demand for help, you should explain what you've done and ask for help - which you might, or might not get depending on peoples time,. availability and experiences (any help you get here is for free). Especially with experimental stuff (but it also applies to everything else).

Be mindful of that and provide necessary data that might allow others to help you.

**Reply by XiaoXin101** [reply]
I use chart version is1 5.0, PostgreSQL chart is AMD64, so I replace it with external PostgreSQL, but The installation of airflow is still wrong, maybe it is still an AMD 64 image, so let's confirm Can 1.5.0 pull the arm image? What adjustments should I make to the configuration

**Reply by potiuk** [reply]
I am not sure if yoy've read what I wrote. Please describe what you've done, provide detailed logs of what does not work - just spend a little more than 2 minutes on investigatig and trying to document your problem.

**Reply by potiuk** [reply]
FYI. I just run the latest 1.6.0rc2 candidate with Airlfow 2.3.0 on ARM (I am testing the RC2 candidate) and I have completely no problem with running it. So Helm + Airlfow 2.3.0 works. No configuration is needed to run it.

You have likely compltely different problem - but honestly - I have no idea what it is. Until you spend time on explaining what it is, there is no way to help you.

**Reply by XiaoXin101** [reply]
pgbouncer_exporter only builds amd64 in 1.5.0, so it can't run on arm, I try to use 1.6.0rc2, I see it has changed

**Reply by pingzh** [reply]
you might not need to use pgbouncer with this two prs:

https://github.com/apache/airflow/pull/22493 and https://github.com/apache/airflow/pull/21565, these two PRs remove unnecessary db connections.
